tools/metrics/metrics.py

```
@@ -0,0 +1,200 @@
+
+def parse_metrics(metrics_text):
+    """
+    Parse Prometheus exposition text into a dict:
+      metric_name → { description, type, unit, labels: [<label_keys>] }
+
+    - Strips empty `{}` on unlabelled samples
+    - Propagates HELP/TYPE from base metrics onto _bucket, _count, _sum
+    - Works regardless of # HELP / # TYPE order
+    - Captures # UNIT metadata if present
+    """
+    import re, logging
```

Ruff (0.8.2)

Redefinition of unused `re` from line 4

Remove definition: `re`

(F811)